### PR TITLE
refactor(1.2-api): migrate to junit5

### PR DESCRIPTION
**Pull request by hulkoba**
Hello 👋

We are from Neighbourhoodie, the implementation partner of the [STF](https://www.sovereigntechfund.de/programs/bug-resilience) Bug Resilience Program. This work is part of our agreed Milestone 1. Upgrade from JUnit 4 to JUnit 5. This PR migrates the tests located in log4j-1.2-api to JUnit5.

Thank you!

## Checklist

* Base your changes on `2.x` branch if you are targeting Log4j 2; use `main` otherwise
* `./mvnw verify` succeeds (if it fails due to code formatting issues reported by Spotless, simply run `./mvnw spotless:apply` and retry)
* Non-trivial changes contain an entry file in the `src/changelog/.2.x.x` directory
* Tests for the changes are provided
* [Commits are signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (optional, but highly recommended)

### CODE REVIEW

**Comment by ppkarwasz** [comment]
@hulkoba,

In #3039 I removed some ancient and disabled tests from `log4j-1.2-api`. To prevent some useless work, make sure you base your PR on a recent `2.x` commit.

**Comment by hulkoba** [comment]
@ppkarwasz This is ready for review now :wave:

File: log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java [hunk 1]
```diff
@@ -98,35 +98,35 @@ public void testForcedLog() {
         category.info("Hello, World");
         List<LogEvent> list = appender.getEvents();
         int events = list.size();
-        assertTrue("Number of events should be 1, was " + events, events == 1);
+        assertTrue(events == 1, "Number of events should be 1, was " + events);
```
Comment (ppkarwasz) [inline log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java:101]:
```suggestion
        assertEquals(1, events, "Number of events");
```

A couple of old assertions did use `assertTrue` to test the equality of two values. Could your replace them with `assertEquals` please?

**Review by ppkarwasz — APPROVED** [review]
LGTM

File: log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java (hunk 1 above)
Comment (ppkarwasz) [reply log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java]: Thanks

**Comment by ppkarwasz** [comment]
@hulkoba, thanks!

**Pull request merged**